Apologies for the delay in looking at this... Life is... Well it's not always easy, huh?

The performance improvements from this look great, but it seems that the test suite fails with a panic:

```
tom@girru:~/src/github.com/Cgboal/unfurl (master)▶ go test
--- FAIL: TestFormat (0.00s)
    --- FAIL: TestFormat/24 (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x505058]

goroutine 32 [running]:
testing.tRunner.func1.2({0x59bc00, 0x70d4c0})
	/usr/local/go/src/testing/testing.go:1389 +0x24e
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1392 +0x39f
panic({0x59bc00, 0x70d4c0})
	/usr/local/go/src/runtime/panic.go:838 +0x207
index/suffixarray.(*Index).lookupAll(0x0, {0xc0000c9d88, 0xb, 0x20})
	/usr/local/go/src/index/suffixarray/suffixarray.go:247 +0x58
index/suffixarray.(*Index).Lookup(0xc000013000?, {0xc0000c9d88?, 0x2?, 0x5c6378?}, 0xffffffffffffffff)
	/usr/local/go/src/index/suffixarray/suffixarray.go:261 +0x56
github.com/Cgboal/DomainParser.(*Parser).FindTldOffset(0x716848, {0xc000012ff0, 0x3, 0x3})
	/home/tom/pkg/mod/github.com/!cgboal/!domain!parser@v0.0.0-20210827145802-99068439e39f/parser.go:45 +0xe5
github.com/Cgboal/DomainParser.(*Parser).GetSubdomain(0xc0000acb40?, {0xc0000169b0?, 0x14?})
	/home/tom/pkg/mod/github.com/!cgboal/!domain!parser@v0.0.0-20210827145802-99068439e39f/parser.go:78 +0x65
github.com/Cgboal/unfurl.extractFromDomain(0xc0000f0d80, {0x5c7ad9, 0x9})
	/home/tom/src/github.com/Cgboal/unfurl/main.go:289 +0xf4
github.com/Cgboal/unfurl.format(0xc0000f0d80, {0x5c63a8, 0x2})
	/home/tom/src/github.com/Cgboal/unfurl/main.go:217 +0x2b7
github.com/Cgboal/unfurl.TestFormat.func1(0xc0000fa000)
	/home/tom/src/github.com/Cgboal/unfurl/format_test.go:63 +0x9c
testing.tRunner(0xc0000fa000, 0xc000071100)
	/usr/local/go/src/testing/testing.go:1439 +0x102
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1486 +0x35f
exit status 2
FAIL	github.com/Cgboal/unfurl	0.005s
```

I'm not sure if that was happening when you raised this PR or not, but I can't merge it in its current state, sorry!

--- a/go.mod
+++ b/go.mod
@@ -2,6 +2,9 @@ module github.com/tomnomnom/unfurl
 
 go 1.18
 
-require github.com/jakewarren/tldomains v0.0.0-20220401175117-e21ec594add9
+require (
+	github.com/Cgboal/DomainParser v0.0.0-20210827145802-99068439e39f
+	github.com/jakewarren/tldomains v0.0.0-20220401175117-e21ec594add9
+)
 
 require golang.org/x/net v0.0.0-20220401154927-543a649e0bdd // indirect

--- a/main.go
+++ b/main.go
@@ -15,6 +15,10 @@ import (
 
 var extractor parser.Parser
 
+func init() {
+	extractor = parser.NewDomainParser()
+}
+
 func main() {
 
 	var unique bool
@@ -51,8 +55,6 @@ func main() {
 	sc := bufio.NewScanner(os.Stdin)
 
 	seen := make(map[string]bool)
-
-	extractor = parser.NewDomainParser()
 
 	for sc.Scan() {
 		u, err := parseURL(sc.Text())
@@ -298,9 +300,9 @@ func extractFromDomain(u *url.URL, selection string) string {
 
 	// remove the port before parsing
 	portRe := regexp.MustCompile(`(?m):\d+$`)
-	
+
 	domain := portRe.ReplaceAllString(u.Host, "")
-	
+
 	switch selection {
 	case "subdomain":
 		return extractor.GetSubdomain(domain)

